### ISSUE TITLE
Discussion: Desired behavior for invalid Unicode escapes

### ISSUE THREAD

**Issue by ChayimFriedman2** [wren-lang/wren#816]
You can create invalid Unicode escapes using `\U`, for example `"\UFFFFFFFF"`.

The current code is problematic from any point of view:

 - It's behavior is not documented
 - It does not keeps them, it just ignores them (`"\UFFFFFFFF".bytes.count == 0` - actually right now it's 1, it's because the bug my PR #814 fixes)
 - It's not errs either

Personally, I tend towards error (if you want raw bytes use `\x`), but keeping them is also an option. Either way, the choice should be documented.

**Comment by mhermier** [comment]
Implementation is lazy and validating it is quite expensive. While I agree
correctness would be good, I wonder if testing it will not introduce to
much code, for even rejecting invalid planes only (since we can't really
rely on external code)

**Comment by ChayimFriedman2** [comment]
No, just another `else` here:

https://github.com/wren-lang/wren/blob/45c67fae0c2fbe78b608d372d0951ffe05f02690/src/vm/wren_compiler.c#L840-L845

**Comment by mhermier** [comment]
I also agree with the arguments. If one really want to introduce binary, `\x` should be used instead of `\u` or `\U`.

This seems to be coherent with what other compiler do. Testing with various languages and compilers (using compiler explorer), it seems that most compiler agree to error at least on code points not being able to be encoded. So I think it is the simplest approach to take.
